### PR TITLE
refactored GET article/:article_id to also return comment count

### PR DESCRIPTION
**Pull request by BrennB99**
- refactored GET article/:article_id to also return comment count.
- Added test to account for this.

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -64,6 +64,7 @@ describe("GET /api/articles/:article_id", () => {
             topic: expect.any(String),
             created_at: expect.any(String),
             votes: expect.any(Number),
+            comment_count: 11,
```
Comment (cjplatten) [inline __tests__/app.test.js]: when you add a new feature you should create a new and separate test, not alter previous tests. If the original test failed with the changes it means it was a brittle test.

File: controllers/app.controllers.js [hunk 2]
```diff
@@ -4,6 +4,7 @@ const {
   retrieveArticle,
   changeArticle,
   retrieveUsers,
+  commentCount,
 } = require("../models/app.models");
```
Comment (cjplatten) [inline controllers/app.controllers.js]: please refactor your controllers and models to separate out the different tables they reference (e.g articles models, topics models).

File: controllers/app.controllers.js [hunk 3]
```diff
@@ -18,8 +19,14 @@ exports.invalidEndpoint = (req, res, next) => {
...
-    .then((article) => {
+  return Promise.all([
+    retrieveArticle(id.article_id),
+    commentCount(id.article_id),
+  ])
+    .then((values) => {
+      const article = values[0];
+      const commentNum = values[1];
+      article.comment_count = commentNum;
       res.status(200).send({ article });
```
Comment (cjplatten) [inline controllers/app.controllers.js]: is there a way to use SQL for this count rather than javascript and two separate database requests?

**Review by cjplatten — CHANGES_REQUESTED** [review]
please see comments for changes to be made and resubmit your pull request.

File: controllers/app.controllers.js (hunk 3 above)
Comment (HeatherEver) [reply controllers/app.controllers.js]: this still hasn't been refactored - think about combining both of these databse requests into one - maybe we can use an aggregate function?

**Review by HeatherEver — CHANGES_REQUESTED** [review]
Just the SQL request to have a look at

**Pull request merged**